tests/unit/test_index.py

```diff
@@ -190,3 +194,85 @@ def test_egg_info_matches(egg_info, search_name, expected):
     link = None     # Only used for reporting.
     version = egg_info_matches(egg_info, search_name, link)
     assert version == expected
+
+
+@pytest.mark.parametrize(
+    "url, vcs_scheme",
+    [
+        ("svn+http://pypi.org/something", "svn"),
+        ("git+https://github.com/pypa/pip.git", "git"),
+    ],
+)
+def test_get_html_page_invalid_scheme(caplog, url, vcs_scheme):
```

Comment:
I think it would help a lot if the ordering of the tests inside the test module can match the ordering of the functions as much as possible. It makes it easier to locate tests and know whether they already exist, etc.

Reply:
Yeah that would make sense. I was walking through the function and write tests for each point an edge case would appear, hence the current order.

Also (not related to this particular comment) I’m moving the tests to its own file because `test_index.py` is getting very long after adding these (and there are more to be added).